src/compiler/checker.ts

```diff
@@ -598,6 +598,7 @@ namespace ts {
             SkipContextSensitive = 1,  // Skip context sensitive function expressions
             Inferential = 2,           // Inferential typing
             Contextual = 3,            // Normal type checking informed by a contextual type, therefore not cacheable
+            NoDeferredCheck = 4,       // Don't add any nodes for deferred checking
         }

```

Comment:
The comment is a little misleading since it only disables deferred checking for class expressions - should probably mention that so someone doesn't come back later and because "it doesn't do what it says it does", "fix" it.